libraries/Microsoft.Bot.Builder/TurnContextServiceCollection.cs

```diff
@@ -65,7 +44,7 @@ public TurnContextServiceCollection()
             }
             catch(ArgumentException)
             {
-                throw new ServiceKeyAlreadyRegisteredException(key);
+                throw new Exception($"A services is already registered with the specified key: {key}");
```

Comment:
I think I'm already on record on this subject, but I strongly disagree that typed exceptions are "useless". I can argue the case broadly, but at a bare minimum: how do you even test this now? Are you going to string check the exception message? What about checking the actual key that was violated? Split the string based on its fragile formatting?

In fact, I'd say a more important problem to be solving is that are no unit tests around this right now. If we had those I think we'd more quickly realize the pain this kind of change inflicts on upstream consumers.

Reply:
+1 to the "No Unit Tests" problem. That's a big problem here.

Reply:
There were no unit tests here.

So the feedback refers to existing code missing unit tests? And not this review?

Certainly agree lack of unit tests is a major issue in this repo.

Reply:
Ultimately the point I was making with the mention of the unit tests was that they are the library's initial "line of defense" in terms of the representation of the experience for users, right? So, in this case, if there was already a unit test and you had changed this code you would quickly realize that you've broken a test and then as you went to rewrite the test you would (hopefully) realize that the removal of the strongly typed exception has created a less than ideal experience for upstream consumers who have to deal with interpreting the exception.